### PR TITLE
fix: update google one-tap docs

### PR DESCRIPTION
**Pull request by kangmingtay**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?
* Update the google one-tap guide configuration and add an example with NextJS 

## What is the current behavior?

Please link any relevant issues here.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: apps/docs/content/guides/auth/social-login/auth-google.mdx [hunk 1]
```diff
@@ -68,7 +68,7 @@ To use Google's pre-built signin buttons:
 1. Go to the [API Credentials page](https://console.cloud.google.com/apis/credentials).
 1. Click `Create credentials` and choose `OAuth Client ID`.
 1. For application type, choose `Web application`.
-1. Under **Authorized JavaScript origins** and **Authorized redirect URLs**, add your site URL. This is the URL of the website where the signin button will appear, _not_ your Supabase project domain.
+1. Under **Authorized JavaScript origins** and **Authorized redirect URLs**, add your site URL. This is the URL of the website where the signin button will appear, _not_ your Supabase project domain. If you're testing in localhost, ensure that you have `http://localhost` set in the **Authorized JavaScript origins** section as well. This is important when integrating with Google One-Tap to ensure that the javascript origin matches.
```
Comment (hf) [inline apps/docs/content/guides/auth/social-login/auth-google.mdx:71]:
```suggestion
1. Under **Authorized JavaScript origins** and **Authorized redirect URLs**, add your site URL. This is the URL of the website where the signin button will appear, _not_ your Supabase project domain. If you're testing in localhost, ensure that you have `http://localhost` set in the **Authorized JavaScript origins** section as well. This is important when integrating with Google One-Tap to ensure you can use it locally.
```

File: apps/docs/content/guides/auth/social-login/auth-google.mdx [hunk 2]
```diff
@@ -341,6 +341,83 @@ Most web apps and websites can utilize Google's [personalized sign-in buttons](h
...
+    const hashBuffer = await crypto.subtle.digest('SHA-256', encodedNonce)
+    const hashArray = Array.from(new Uint8Array(hashBuffer))
+    const hashedNonce = hashArray.map((b) => b.toString(16).padStart(2, '0')).join('')
+
+    return [nonce, hashedNonce]
+  }
+
+  useEffect(() => {
+    const initializeGoogleOneTap = () => {
+      window.onload = async () => {
```
Comment (hf) [inline apps/docs/content/guides/auth/social-login/auth-google.mdx:374]: Hmm why is this `window.onload` necessary? Prefer using `addEventListener` in `useEffect` so you can remove it as React in strict mode runs the effect twice. 😭

File: apps/docs/content/guides/auth/social-login/auth-google.mdx [hunk 3]
```diff
@@ -341,6 +341,83 @@ Most web apps and websites can utilize Google's [personalized sign-in buttons](h
...
+              console.log('Successfully logged in with Google One Tap')
+
+              // redirect to protected page
+              router.push('/')
+            } catch (error) {
+              console.error('Error logging in with Google One Tap', error)
+            }
+          },
+          nonce: hashedNonce,
+          use_fedcm_for_prompt: true,
```
Comment (J0) [inline apps/docs/content/guides/auth/social-login/auth-google.mdx:402]: This option was a little puzzling for me, can we consider adding a note adding an explanation or a reference to the section where [the use of the flag is described](https://supabase.com/docs/guides/auth/social-login/auth-google#google-pre-built)?

>To make your app compatible with Chome's third-party-cookie phase-out, make sure to set data-use_fedcm_for_prompt to true.

**Review by J0 — APPROVED** [review]
Nice work

**Pull request merged**